config/routes.rb

```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
   root to: 'users#index'
-  get 'users', to: 'users#index'
-  get 'users/:id', to: 'users#show'
-  get 'users/:id/posts', to: 'posts#index'
-  get 'users/:id/posts/:id', to: 'posts#show'
+  
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create] 
+      resources :likes, only: [:create, :destroy]
```

Comment:
:red_circle:  noticed that you've defined routes for the likes resource, but you haven't created a controller to handle those routes. This is not a good practice for a few reasons:
 - It's not RESTful. REST is a set of architectural principles for designing web APIs. One of the principles of REST is that each resource should have its own controller. By not creating a controller for the likes resource, you're violating this principle.
 - It makes it difficult to test the likes feature. Without a controller, you won't be able to use Rails' built-in testing framework to test the logic for creating and destroying likes.